### DISCUSSION TITLE
Role of hyperparameters in 20 second regression version

### DISCUSSION THREAD

**Discussion by patel-zeel** [wesselb/stheno#9] [Q&A]
I was a bit curious about what is going on in the background when we compute posterior in the [20-second version](https://github.com/wesselb/stheno#nonlinear-regression-in-20-seconds). Is there any hyperparameter tuning happening in the background? Or it is just computing the posterior with default hyperparameters. 

To put it in another way, [this version](https://github.com/wesselb/stheno#nonlinear-regression-in-20-seconds) is ready to use in production or I need to use some additional steps to get a better posterior?

**Comment by wesselb** [comment]
Hey!

To directly answer your question, in the 20-second version, there is no hyperparameter tuning happening. It is just computing the posterior with the default hyperparameters (variance 1 and length scale also 1). If you'd like to do something like this in production, I would recommend first doing maximum likelihood to get a better estimate of the length scale and variance.

**Reply by patel-zeel** [reply]
Hi, Thanks for solving the query. I got too excited at first sight and deployed it for real-world application before I realized this :)

**Reply by wesselb** [reply]
No problem at all! I'm glad to hear that you got excited. :) If there is a need for a way to more easily and succinctly incorporate hyperparameter learning, then we could certainly work something out!

**Reply by patel-zeel** [reply]
Sure. Currently, I find it neat and clean but we can look into this as I explore more. 

I must mention that you have an amazing set of decentralized tools working together (Stheno, mlkernels, varz, lab and what not!).
